src/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/ChunkedEncodingReadStream.cs

```diff
@@ -300,8 +307,14 @@ private ReadOnlyMemory<byte> ReadChunkFromConnectionBuffer(int maxBytesToRead, C
                                     _state = ParsingState.Done;
                                     _connection.CompleteResponse();
                                     _connection = null;
+
                                     break;
                                 }
+                                // Parse the trailers.
+                                else
```

Comment:
This `else` is where we should avoid parsing trailers if we're draining the connection, e.g. `else if (!IsDisposed)`.

Reply:
Thanks, will fix.